src/managed/Microsoft.DotNet.Build.Bundle/README.md

```diff
@@ -0,0 +1,13 @@
+.NET Core Bundler
+===================================
+
+The Bundler is a tool that embeds an application and its dependencies into the AppHost executable. This tool is used to publish apps as a single-file, as described in this [design document](https://github.com/dotnet/designs/blob/master/accepted/single-file/design.md).
+
+### Why is the Bundler in core-setup repo?
```

Comment:
I feel like a tool vs. library section would be useful too. Draft from the thread:

---

### Why is the Bundler a tool, not a managed library?

The user-facing scenario is `dotnet publish ...`, implemented by MSBuild tasks, however the Bundler is an executable tool with a command-line interface. This is done for several reasons:

1. A library requires extra effort to ensure it can be hosted by the MSBuild process.
   * Target framework multitargeting?
   * Will dependencies loaded to run this library interfere with the dependencies other tasks use, or vice versa?
1. Can be run without spinning up a full build in "ad-hoc" scenarios.
1. Makes testing easier.
1. Forces a crisp contract of inputs and outputs.

The [IL Linker](https://github.com/mono/linker) is similarly implemented as an out-of-process tool.